packages/xo-server/src/xapi/index.js

```diff
@@ -1111,25 +1097,15 @@ export default class Xapi extends XapiBase {
       force = false,
     }
   ) {
-    const getDefaultSrRef = once(() => {
-      if (sr !== undefined) {
-        return hostXapi.getObject(sr).$ref
-      }
-      const defaultSr = host.$pool.$default_SR
-      if (defaultSr === undefined) {
-        throw new Error(`This operation requires a default SR to be set on the pool ${host.$pool.name_label}`)
-      }
-      return defaultSr.$ref
-    })
-
     // VDIs/SRs mapping
+    const srRef = hostXapi.getObject(sr).$ref
     const vdis = {}
     const vbds = flatMap(vm.$snapshots, '$VBDs').concat(vm.$VBDs)
     for (const vbd of vbds) {
       const vdi = vbd.$VDI
       if (vbd.type === 'Disk') {
         vdis[vdi.$ref] =
-          mapVdisSrs && mapVdisSrs[vdi.$id] ? hostXapi.getObject(mapVdisSrs[vdi.$id]).$ref : getDefaultSrRef()
+          mapVdisSrs && mapVdisSrs[vdi.$id] ? hostXapi.getObject(mapVdisSrs[vdi.$id]).$ref : srRef ?? vdi.$SR.$ref
```

Comment:
I believe the logic should be:
1. explicit SR for this VDI
2. if VDI must be migrated (current SR not reachable from host)
    1. default SR for this migration
    2. default SR for this pool